actors/power/src/lib.rs

```diff
@@ -129,7 +129,7 @@ impl Actor {
                 },
             )?;
             st.miner_count += 1;
-
+            st.add_pledge_total(rt.policy().new_miner_deposit.clone());
```

Comment:
This code is a long way from the other code that it has to stay in sync with. This has a high chance of introducing an error over time. Instead, I think the miner should notify the power actor of the locked tokens the same way it does for every other locking: by invoking the update pledge method. We might need to re-arrange this method to ensure the state is ready for the re-entrant call and will not be overwritten.

Reply:
It was initially intended to be written this way, but the current method doesn't support it. The behavior of the existing method had to be modified.

```rs

    fn update_pledge_total(
        rt: &impl Runtime,
        params: UpdatePledgeTotalParams,
    ) -> Result<(), ActorError> {
        rt.validate_immediate_caller_type(std::iter::once(&Type::Miner))?;
        rt.transaction(|st: &mut State, rt| {
            st.validate_miner_has_claim(rt.store(), &rt.message().caller())?;
            st.add_pledge_total(params.pledge_delta);
            if st.total_pledge_collateral.is_negative() {  // NOTICE need allow negative value in this line
                return Err(actor_error!(
                    illegal_state,
                    "negative total pledge collateral {}",
                    st.total_pledge_collateral
                ));
            }
            Ok(())
        })
    }
```